lib/AST/Identifier.cpp

```diff
@@ -20,13 +20,21 @@
 #include "llvm/Support/ConvertUTF.h"
 using namespace swift;
 
+void *DeclBaseName::SubscriptIdentifierData =
```

Comment:
Casting this to `char *` to initialize an Identifier is technically illegal. Maybe DeclBaseName should just store a `void *` and avoid the whole mess?

(Also, it would be nice if it actually *were* a string, so that you could get something useful in the debugger. But we'd have to make sure that string was 4-byte aligned.)